src/main/java/org/tdl/vireo/model/repo/impl/SubmissionRepoImpl.java

```diff
@@ -745,6 +772,77 @@ public int compare(SubmissionListColumn svc1, SubmissionListColumn svc2) {
+                    sqlBuilder = new StringBuilder();
+                    if (!sqlJoinsBuilder.toString().contains("LEFT JOIN action_log al ON al.id=s.last_action_id")) {
+                        sqlBuilder.append("\nLEFT JOIN action_log al ON al.id=s.last_action_id");
+                    }
+
+                    sqlJoinsBuilder.append(sqlBuilder);
+                    sqlCountSelectBuilder.append(sqlBuilder);
+
+                    if (submissionListColumn.getSortOrder() > 0) {
+                        setColumnOrdering(submissionListColumn.getSort(), sqlAliasBuilders, sqlOrderBysBuilder, "al.entry");
+                    }
+
+                    for (String filterString : submissionListColumn.getFilters()) {
+                        sqlBuilder = new StringBuilder();
+
+                        if (filterString == null) {
+                            sqlBuilder.append("al").append(".entry IS NULL");
+                        } else if (submissionListColumn.getExactMatch()) {
+                            sqlBuilder.append("al").append(".entry = '").append(filterString).append("'");
+                        } else {
+                            sqlBuilder.append("LOWER(al").append(".entry) LIKE '%").append(escapeString(filterString)).append("%'");
+                        }
+
+                        sqlWhereBuilderList.add(sqlBuilder);
+                        getFromBuildersMap(sqlCountWhereFilterBuilders, "lastAction.entry").add(sqlBuilder);
+                    }
+
+                    // all column search filter
+                    for (String filterString : allColumnSearchFilters) {
+                        sqlBuilder = new StringBuilder();
+                        sqlBuilder.append("LOWER(al").append(".entry) LIKE '%").append(escapeString(filterString)).append("%'");
+                        sqlAllColumnsWhereBuilderList.add(sqlBuilder);
+                    }
+
+                    break;
+
+                case "lastAction.actionDate":
+                    sqlBuilder = new StringBuilder();
+                    if (!sqlJoinsBuilder.toString().contains("LEFT JOIN action_log al ON al.id=s.last_action_id")) {
+                        sqlBuilder.append("\nLEFT JOIN action_log al ON al.id=s.last_action_id");
+                    }
+
+                    sqlJoinsBuilder.append(sqlBuilder);
+                    sqlCountSelectBuilder.append(sqlBuilder);
+
+                    if (submissionListColumn.getSortOrder() > 0) {
+                        setColumnOrdering(submissionListColumn.getSort(), sqlAliasBuilders, sqlOrderBysBuilder, "al.action_date");
+                    }
+
+                    for (String filterString : submissionListColumn.getFilters()) {
+                        if (filterString.contains("|")) {
+                            String[] dates = filterString.split(Pattern.quote("|"));
+                            sqlBuilder = new StringBuilder()
+                                .append("al.").append("action_date")
```

Comment:
Why not?
```
.append("al.action_date")
```

(And such in all similarly written added code.)

Reply:
Indeed. Why not?

Reply:
Done.